refactor(common): remove doubles of the word "the" in comments

Vicachu42 opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
Found duplicates of the word "the" in several comments.

Issue Number: N/A


## What is the new behavior?
Removed the duplicates.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


## Other information

mmalerba committed:
feat(material/table): pass `this` rather than concrete classes in the harnesses (#25482)

some users are extending the harnesses, and passing the concrete class
broke them

ko-tori committed:
fix(cdk/overlay): backdropClass type mismatch (#25487)

wagnermaciel committed:
switch progress-spinner implementation to use MDC (#25324)

* feat(material/progress-spinner): 1. Move src/material/progress-spinner to src/material/legacy-progress-spinner

* feat(material/progress-spinner): 2. rename ts symbols in src/material/legacy-progress-spinner

* feat(material/progress-spinner): 3. rename sass mixins in src/material/legacy-progress-spinner

* feat(material/progress-spinner): 4. move src/material-experimental/mdc-progress-spinner to src/material/progress-spinner

* feat(material/progress-spinner): 5. update the all-theme mixins

* fix(material/progress-spinner): fix lint errors

* fix(material/progress-spinner): fix build failures after rebase

* fixup! fix(material/progress-spinner): fix build failures after rebase

* fixup! fix(material/progress-spinner): fix build failures after rebase

* fixup! fix(material/progress-spinner): fix build failures after rebase

* fixup! fix(material/progress-spinner): fix build failures after rebase

* fixup! fix(material/progress-spinner): fix build failures after rebase

* fixup! fix(material/progress-spinner): fix build failures after rebase

mmalerba committed:
feat(material/core): move MDC-based elevation styles out of experimental (#25486)

* feat(material/core): move MDC-based elevation styles out of experimental

* handle CSS var elevation colors

* undo rearranging core theme mixins, since it broke some things

* undo forwarding mdc-core-color to core-color

* fix lint issues

andrewseguin committed:
docs: release notes for the v14.1.3 release

andrewseguin committed:
docs: release notes for the v14.2.0-rc.0 release

andrewseguin committed:
feat(material/table): migrate MDC examples (#25492)

andrewseguin committed:
feat(material/tooltip): migrate MDC examples (#25495)

amysorto committed:
feat(material/schematics): add template migration support within typescript files (#25496)

andrewseguin committed:
feat(material/paginator): migrate MDC examples (#25497)

* feat(material/paginator): migrate MDC examples

* feat(material/paginator): migrate MDC examples

crisbeto committed:
build: rename legacy examples

Renames the legacy component examples to be consistent.

crisbeto committed:
build: fix up legacy paginator demo

Removes the legacy paginator examples since they won't be used for
anything.

crisbeto committed:
build: fix up legacy table demo

Fixes up the file names of the legacy table dev app examples and moves the
docs examples into the dev app.

crisbeto committed:
build: fix up legacy tooltip demo

Fixes the file and class names in the legacy tooltip demo and moves the
examples into the dev app.

crisbeto committed:
build: update to TypeScript 4.8 (#25505)

Updates the repo to TypeScript 4.8 and resolves some of the breaking changes. I've only kept the schematics code backwards-compatible since the rest of the changes are to our own tooling.

crisbeto committed:
build: migrate MDC input demo (#25506)

Renames the current input demo to `legacy` and moves the MDC version to stable.

crisbeto committed:
build: switch select demo app to MDC

Switches the `mat-select` demo app to use MDC and fixes that the docs were using some legacy components.

crisbeto committed:
docs(material/input): switch to MDC

Finishes switching the input docs and demo app to MDC. The docs examples themselves were switched already, but these changes finish it up by:
* Fixing some imports that were still pointing to legacy components.
* Removing the explicit `appearance` since it's not longer necessary.
* Deleting the MDC-specific custom form control example since it wasn't being used anywhere and the one for the legacy component still applies.
* Using `matNativeControl` instead of `matInput`.

crisbeto committed:
build: remove legacy component overviews (#25509)

Since we won't be publishing docs for the legacy components, we don't need to keep the overviews around.

angular-robot committed:
build: update dependency @bazel/bazelisk to v1.12.1 (#25475)

devversion committed:
build: invalidate circleci cache to prune unused nested node modules (#25512)

Yarn 1.x has a bug where it keeps nested unused node modules and doesn't
delete them automatically. This throws off Bazel in some scenarios when
the lock file is updated. This commit invalidates the cache to get a
fresh clean node modules cache without any unused nested directories.

mmalerba committed:
fix(material/core): Move remaining MDC core functionality out of experimental (#25503)

wagnermaciel committed:
fix(material/schematics): make ts import replacements more specific (#25504)

* fix(material/schematics): make ts import replacements more specific

* fixup! fix(material/schematics): make ts import replacements more specific

* fixup! fix(material/schematics): make ts import replacements more specific

EmmanuelRoux committed:
feat(cdk/a11y): add named export and public property to CdkMonitorFocus directive (#25427)

crisbeto committed:
fix(cdk/scrolling): error during server-side rendering (#25461)

Fixes that an error was being thrown by the virtual scroller during server-side rendering. It wasn't failing the SSR check, because the error happens inside of a promise callback.

mmalerba committed:
docs(material/snack-bar): update docs & examples for MDC-based snackbar (#25514)

* docs(material/snack-bar): update docs & examples for MDC-based snackbar

* update dev-app

angular-robot committed:
build: update angular shared dev-infra code to 6a13e27 (#25394)

wagnermaciel committed:
fix(material/schematics): make ts symbol renamings more specific (#25522)

* fix(material/schematics): make ts symbol renamings more specific

* fixup! fix(material/schematics): make ts symbol renamings more specific

* fixup! fix(material/schematics): make ts symbol renamings more specific

* fixup! fix(material/schematics): make ts symbol renamings more specific

mmalerba committed:
docs(material/card): update docs & examples for MDC-based card (#25517)

* docs(material/card): update docs & examples for MDC-based card

* delete mdc example

* update dev-app

* fix e2e tests

* update docs markdown

wagnermaciel committed:
fix(material/schematics): add renaming of non standard symbols (#25523)

wagnermaciel committed:
feat(material/schematics): add handling for all-component-themes (#25528)

andrewseguin committed:
fix(material/table): cleanup legacy usage (#25525)

josephperrott committed:
build: update ng-dev version (#25529)

update to the latest version of ng-dev

amysorto committed:
docs: release notes for the v14.2.0 release

devversion committed:
build: remove exceptional minor logic (#25535)

We no longer have an exceptional minor and can remove the logic we put
in place given the release tool not supporting exceptional minors well
yet.

crisbeto committed:
build: update to TypeScript 4.8 final (#25536)

Updates to the latest stable version of TS and removes some workaround.

crisbeto committed:
fix(cdk/collections): ConnectableObservable incorrectly inferred as data source (#25526)

The `isDataSource` function determines that something is a data source by checking if it has a `connect` method. The problem is that rxjs has a `ConnectableObservable` which also has a `connect` method and which gets incorrectly identified as a data source.

Fixes #25483.

wagnermaciel committed:
refactor(material/legacy-autocomplete): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-button): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-card): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-checkbox): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-chips): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-dialog): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-form-field): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-input): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-list): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-menu): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-paginator): rename ts symbols

zarend committed:
fix(material/datepicker): calendar aria-descriptions start/end date (#25457)

For date ranges, add aria-descriptions to the cell of the current start
date and also for end date.

Popuplate aria descriptions with the expected value of the ARIA accessible name of the
`matStartDate` and `matEndDate` inputs. Introduces `_computeAriaAccessibleName` function to
implement ARIA acc-name-1.2 specificiation.

Fixes angular#23442 and angular#23445

crisbeto committed:
fix(material/button-toggle): incorrect event source in some cases (#25544)

Removes some logic that was trying to incorrectly match the button toggle being clicked with the model value. The logic wasn't necessary since we already know which toggle the user interacted with.

Fixes #25489.

crisbeto committed:
fix(multiple): resolve extended template diagnostic warnings (#25546)

Turns the extended template diagnostic warnings into errors and resolves all of the existing ones.

crisbeto committed:
docs(material/checkbox): switch docs and examples to MDC (#25547)

Switches the checkbox live examples and docs to use MDC.

crisbeto committed:
docs(material/list): switch examples and demo to MDC (#25520)

Switches the list live examples and demo to use MDC. Also updates the docs to reflect the latest API.

mgoldberg524 committed:
test(cdk/testing): add includeAsyncTests flag (#25542)

* test(cdk/testing): add includeAsyncTests flag

This is needed by some downstream custom angular harnesses that use this set of tests but also do not rely on Angular framework stabilization.

* test(cdk/testing): rework `includeAsyncTests` into `skipAsyncTests`

* test(cdk/testing): fix stale param doc

mmalerba committed:
fix(material/dialog): allow customizing animation duration (#25524)

* fix(material/dialog): allow customizing animation duration

BREAKING CHANGE:
- Passing strings for MatDialogConfig.enterAnimationDuration and
  MatDialogConfig.exitAnimationDuration is deprecated, pass numbers
  in ms instead

* deprecate the string animation durations in favor of numbers

andrewseguin committed:
refactor: rename legacy components in e2e (#25549)

* refactor: rename legacy components in e2e

* ci: fix test

* ci: fix button test

andrewseguin committed:
feat(material/datepicker): expose datepicker symbols (#25552)

wagnermaciel committed:
refactor(material/legacy-progress-bar): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-progress-spinner): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-radio): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-select): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-slide-toggle): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-slider): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-snack-bar): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-table): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-tabs): rename ts symbols

wagnermaciel committed:
refactor(material/legacy-tooltip): rename ts symbols

angular-robot[bot] committed:
build: update `SECURITY.md` to match the content of `angular/.github`

amysorto committed:
fix(material/schematics): update theme names for MDC migration schematic (#25550)

mmalerba committed:
feat(material/core): make mdc-based typography default (#25551)

mmalerba committed:
docs(material/dialog): update examples to use MDC-based dialog (#25560)

crisbeto committed:
fix(material/legacy-form-field): remove invalid selector (#25554)

We used to use `:placeholder-shown` to toggle some styles in the form field. In #8223 those selectors were changed to `:label-shown` which isn't a valid CSS selector.

These changes remove the selector since it can cause errors in some tools.

Fixes #24181.

jermowery committed:
Use more specific MDC slider targets (#25562)

Switches to use more specific MDC slider targets to fix internal test failures

zarend committed:
build: fix lint error in slider (#25570)

Fix lint error in slider by running `yarn buildifier
src/material/slider/BUILD.bazel`

mgoldberg524 committed:
test(cdk/testing): Allow TestComponentsModule to be bootable as a module for a standalone app (#25561)

This is required by downstream tests that use the cross-environment tests but do not start up the entire e2e test app in this repo. This change will be copybara'ed down to google3

amysorto committed:
docs(material/button): switch examples and demo to MDC (#25559)

amysorto committed:
docs(material/menu): switch examples and demo to MDC (#25571)

amysorto committed:
docs(material/radio): switch examples and demo to MDC (#25572)

mmalerba committed:
fix(material/legacy-autocomplete): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-button): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-card): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-checkbox): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-chips): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-core): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-dialog): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-form-field): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-input): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-list): deprecate all ts symbols

wagnermaciel committed:
fix(material/legacy-menu): rename missed ts symbol

wagnermaciel committed:
build: add checks for public-api.ts formats (#25586)

* check that all exports from public-api.ts are  explicit exports
* check that all exported symbols contain 'legacy'

mmalerba committed:
fix(material/legacy-menu): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-paginator): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-progress-bar): deprecate all ts symbols

wagnermaciel committed:
fix(material/legacy-select): rename missed ts symbol

mmalerba committed:
fix(material/legacy-progress-spinner): deprecate all ts symbols

wagnermaciel committed:
fix(material/legacy-tooltip): rename missed ts symbol

mmalerba committed:
fix(material/legacy-radio): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-select): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-slide-toggle): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-slider): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-snack-bar): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-table): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-tabs): deprecate all ts symbols

mmalerba committed:
fix(material/legacy-tooltip): deprecate all ts symbols

mmalerba committed:
fix(multiple): deprecate all legacy sass mixins

wagnermaciel committed:
build: tidy up public-api script output (#25588)

wagnermaciel committed:
docs: release notes for the v14.2.1 release

wagnermaciel committed:
release: update Angular version range (#25595)

wagnermaciel committed:
release: cut the v15.0.0-next.0 release

mmalerba committed:
fix(material/progress-bar): add coercion for number inputs (#25585)

* fix(material/progress-bar): add coercion for number inputs

* fixup! fix(material/progress-bar): add coercion for number inputs

mmalerba committed:
fix(material/card): remove card deps on legacy-card

mmalerba committed:
fix(material/dialog): remove dialog deps on legacy-dialog

mmalerba committed:
fix(material/slide-toggle): remove slide-toggle deps on legacy-slide-toggle

mmalerba committed:
fix(material/snack-bar): fix legacy reference in docs

mmalerba committed:
fix(material/table): fix legacy reference in docs

mmalerba committed:
fix(material/tabs): remove tabs deps on legacy-tabs

mmalerba committed:
fix(material/tooltip): remove tooltip deps on legacy-tooltip

crisbeto committed:
fix(material/button): remove dependency on legacy button (#25579)

Fixes that the non-legacy button was depending on the legacy button.

crisbeto committed:
fix(material/progress-bar): remove dependency on legacy progress bar (#25581)

Fixes that the non-legacy progress bar was importing several symbols from the legacy version.

crisbeto committed:
fix(material/progress-spinner): remove dependency on legacy progress spinner (#25583)

Fixes that the non-legacy progress spinner was depending on the legacy version.

crisbeto committed:
build: remove MDC exports script

The MDC exports script isn't being used anymore.

crisbeto committed:
build: remove MDC tests script

The `check-mdc-tests` script isn't being used anymore.

mmalerba committed:
fix(material/paginator): remove deps on legacy components

mmalerba committed:
fix(material/sort): remove deps on legacy components

mmalerba committed:
fix(material/stepper): remove deps on legacy components

XuankangLin committed:
docs: fix comment style by adding missing periods. (#25606)

mmalerba committed:
fix(material/core): fix swapped units in define-typography functions (#25615)

crisbeto committed:
fix(cdk/listbox): error with multiple preselected values (#25621)

The CDK listbox had a subscription to its model changes in the constructor which fires immediately due to the `startsWith` and which was attempting to sort the selected values when in multiple selection mode. The problem is that the options aren't available until `ngAfterContentInit` which was causing an error when we tried to do the sorting.

These changes move the change event subscription to `ngAfterContentInit` so that options are guaranteed to be defined when it fires.

Fixes #25611.

crisbeto committed:
fix(cdk/menu): don't prevent default enter and space actions (#25591)

Fixes that the CDK menu trigger and menu item were preventing the default action on enter and space key presses. Presumably this was done to avoid form submissions, but it can also prevent anchors from being used as menu items. These changes avoid submissions by setting the `type` attribute on `button` elements.

Fixes #25584.

crisbeto committed:
fix(material/menu): update submenu indication when menu is assigned (#25608)

In #24437 we added the ability to conditionally remove a menu from a menu trigger. The problem is that the flag that determines if the menu item has the small triangle indicator is only set in the constructor. These changes move the logic to the setter and add a `markForCheck` to ensure that the view is in sync.

crisbeto committed:
refactor(material/core): fix deprecation warning (#25622)

Fixes a deprecation warning, because we were using a slash for division instead of `math.div`.

angular-robot committed:
build: update angular shared dev-infra code to 50647a2 (#25521)

crisbeto committed:
fix(cdk/a11y): account for Windows 11 high contrast themes in detector (#25624)

Windows 11 has different built in high contrast themes which weren't being picked up by the detector.

Fixes #25580.

crisbeto committed:
fix(material/datepicker): restore focus after closing animation (#25567)

Currently the datepicker restores focus when the closing sequence starts, but in some cases that can cause focus to be lost. When the calendar view changes, we delay moving focus for a while in order to work around a VoiceOver issue. This means that if a closing sequence is kicked off while there's an in-progress view change, focus may end up being restored, then moved back into the calendar which then gets destroyed. This can be seen in the "Datepicker emulating a Year and month picker" example.

These changes resolve the issue by moving the focus restoration all the way to the end of the closing sequence.

Fixes #25564.

crisbeto committed:
refactor(material/progress-bar): switch to new theming API (#25576)

Switches the progress bar component to the new theming API from MDC.

andrewseguin committed:
fix(material/core): warn when legacy theme is created (#25625)

* fix(material/core): warn when legacy theme is created

* fix(material/core): update message

* fix(material/core): update message

* fix(material/core): update message

* fix(material/core): update message

* fix(material/core): update message

crisbeto committed:
docs(material/slider): switch examples to use MDC

Switches the slider examples to use the MDC slider. Also updates the guide and adds an example of a range slider.

I also ended up removing the slider from the "Getting started" guide since the new API is a bit more complicated than the old one.

crisbeto committed:
build: switch slider demo to use MDC

Switches the demo for the slider to use the MDC implementation.

crisbeto committed:
build: update to Angular 15 (#25628)

Updates the repo to Angular 15 and fixes some build errors.

andrewseguin committed:
docs(material/progress-bar): switch examples to MDC (#25629)

andrewseguin committed:
docs(material/progress-spinner): switch examples to MDC (#25630)

mmalerba committed:
fix(material/form-field): fix typography for native input element (#25616)

amysorto committed:
fix(material/schematics): update and handle import statements for the MDC migration

zarend committed:
docs(material/tabs): switch examples and demo to MDC (#25557)

Switches the tabs live examples and demo to use MDC.

crisbeto committed:
docs(material/autocomplete): switch docs to use MDC

Changes the docs for the autocomplete to use MDC implementation.

crisbeto committed:
build: switch autocomplete demo use MDC

Switches the demo for the autocomplete to use the MDC implementation.

crisbeto committed:
build: fix dev app sidenav links

Fixes that there was an incorrect link in the dev app sidenav and that the tabs link wasn't sorted.

angular-robot committed:
build: update angular shared dev-infra code to 6f2cb5c (#25623)

devversion committed:
ci: re-enable until front of line blocking in publish snapshot job (#25602)

We had to temporarily disable the CircleCI orb command that helps
ensuring snapshot jobs do not conflict when running concurrently.

It seems like the CircleCI API has been fixed to no longer cause
issues with the Orb. This commit attempts to re-enable the command.

mmalerba committed:
fix(material/select): fix arrow alignment (#25626)

mmalerba committed:
docs: release notes for the v14.2.2 release

mmalerba committed:
release: cut the v15.0.0-next.1 release

angular-robot[bot] committed:
build: update `CODE_OF_CONDUCT.md` to match the content of `angular/.github`

crisbeto committed:
test(material/select): attempt to reduce test flakes (#25646)

Tries to reduce the amount of flakes we've been seeing in the select tests by getting one of the slower tests to run faster.

zarend committed:
docs(material/slide-toggle): migrate examples to MDC (#25643)

Migrate slide-toggle example to use mdc-based components.

crisbeto committed:
build: switch remaining dev app demos to MDC (#25649)

Switches all of the remaining demos of components that don't have MDC counterparts to use MDC components.

crisbeto committed:
docs: switch remaining docs examples to MDC (#25653)

Switches the remaining usages of legacy components in other components' examples to use the non-legacy versions.

andrewseguin committed:
fix(material/core): add optional warning for incomplete themes (#25654)

andrewseguin committed:
fix(material/core): use full theme config definitions to prebuilt themes (#25656)

mmalerba committed:
docs: add an MDC migration guide (#25543)

* docs: add an MDC migration guide

* fixup! docs: add an MDC migration guide

* fixup! docs: add an MDC migration guide

* fixup! docs: add an MDC migration guide

crisbeto committed:
Switch progress bar and progress spinner docs to MDC (#25659)

* docs(material/progress-bar): switch progress bar docs to use MDC

Switches the progress bar docs to use the MDC components.

* docs(material/progress-spinner): switch progress spinner docs to use MDC

Switches the progress spinner docs to use the MDC components.

crisbeto committed:
feat(material-luxon-adapter): update to luxon 3 (#25660)

Updates the Luxon date adapter to Luxon 3 and fixes a compilation error.

BREAKING CHANGE:
The `MatLuxonDateModule` now requires at least version 3 of Luxon.

Fixes #25408.

zarend committed:
fix(material/datepicker): announce the "to" when reading year range (#24958)

Create period button's `aria-description` using `formatYearRangeLabel`
method. Format year range in a TTS friendly way (e.g. "2019 to 2020").
Previously, some screen readers would announce the range as "2019 2020".

Fixes #23467.

mmalerba committed:
fix(material/schematics): add missing legacy TS symbol transforms

mmalerba committed:
fix(material/schematics): update typescript symbols in test code

mmalerba committed:
fix(material/schematics): add missing legacy sass mixin transforms

mmalerba committed:
fix(material/schematics): update sass function names

pkozlowski-opensource commented:
@Vicachu42 I'm surprised to see the `.yarn/releases/yarn-1.22.17.cjs` file changes as part of this PR, this seems to have some formatting changes and was included by accident. Could you please remove?

crisbeto committed:
docs: delete unused file (#25665)

Removes a file that appears to have been copied by accident.

Vicachu42 commented:
@pkozlowski-opensource Thank you, good catch! It is amended now.

crisbeto committed:
build: switch dev app layout to use MDC (#25670)

Switches the dev app layout to use the MDC versions of the components and fixes a couple of visual issues.

crisbeto committed:
fix(material/stepper): remove invalid aria attribute (#25644)

We have an `aria-expanded` attribute on the `tabpanel` which is invalid. See https://w3c.github.io/aria/#aria-expanded.

Fixes #25637.

crisbeto committed:
test(cdk/schematics): account for breaking change in CLI (#25674)

The CLI started throwing an error if a projects doesn't have a `root` property. These changes update a few of our tests that were failing because of it.

forsti0506 committed:
feat(material/select): add page down/up button functionality (#25508)

crisbeto committed:
build: switch button e2e to MDC

Removes the e2e app entry for the legacy button and renames the MDC button.

crisbeto committed:
build: switch checkbox e2e to MDC

Removes the e2e app entry for the legacy checkbox and renames the MDC checkbox.

crisbeto committed:
build: switch dialog e2e to MDC

Removes the e2e app entry for the legacy dialog and renames the MDC dialog.

crisbeto committed:
build: switch input e2e to MDC

Removes the e2e app entry for the legacy input and renames the MDC input.

crisbeto committed:
build: delete MDC chips e2e demo

The MDC chips e2e demo wasn't being used for anything so this commit deletes it.

crisbeto committed:
build: switch menu e2e to MDC

Removes the e2e app entry for the legacy menu and renames the MDC menu.

crisbeto committed:
build: switch progress bar e2e to MDC

Removes the e2e app entry for the legacy progress bar and renames the MDC progress bar.

crisbeto committed:
build: switch progress spinner e2e to MDC

Removes the e2e app entry for the legacy progress spinner and renames the MDC progress spinner.

crisbeto committed:
build: switch radio e2e to MDC

Removes the e2e app entry for the legacy radio and renames the MDC radio.

crisbeto committed:
build: switch slide toggle e2e to MDC

Removes the e2e app entry for the legacy slide toggle and renames the MDC slide toggle.

crisbeto committed:
build: switch tabs e2e to MDC

Removes the e2e app entry for the legacy tabs and renames the MDC tabs.

crisbeto committed:
build: rename MDC slider e2e demo

Renames the MDC slider e2e demo the remove the MDC prefix. Also adds the slider demo to the navigation.

crisbeto committed:
build: rename MDC table e2e demo

Renames the MDC table e2e demo to remove the MDC prefix.

crisbeto committed:
build: remove remaining legacy references in e2e

Removes the remaining usages of legacy components in the e2e app. Also adds a click handler so the navigation closes when a link is clicked.

mmalerba committed:
fix(material/legacy-core): name all ts symbols to reflect legacy-ness (#25673)

mmalerba committed:
fix(material/schematics): split core imports in ng update (#25678)

Some of the imports from @angular/material/core (the ones related to
MatOption) need to be migrated to import from
@angular/material/legacy-core. However, the remaining symbols should
still import from @angular/material/core. This PR changes the ng update
script to separate them into separate import statements.

amysorto committed:
fix(material/schematics): add support for replacing mixins that begin with all-legacy-component

amysorto committed:
fix(material/schematics): improve sass comment adding logic

- Refactor the code to be more reusable
- Mark comments with TODO(mdc-migration)

devversion committed:
build: use shared circleci dev-infra for common circleci code (#25651)

This allows us to remove the duplicated `rebase-pr.js` script so that we
can maintain this script in a single place.

devversion committed:
build: update shared dev-infra orb to fix snapshot tests (#25690)

Contains: https://github.com/angular/dev-infra/pull/838

amysorto committed:
fix(material/schematics): add custom replacements for imports

devversion committed:
build: setup remote execution through dev-infra Orb (#25694)

andrewseguin committed:
fix(material/core): add flag whether to copy color theme values (#25695)

* fix(material/core): add flag whether to copy color theme values

* fix(material/core): remove legacy theming in devapp

crisbeto committed:
fix(material/dialog): aria-modal not being set

Fixes that `aria-modal` wasn't being set on the dialog container. We had a host binding for it already, but the config didn't have an `ariaModal` property. This likely got broken when the dialog was changed to be based on top of the CDK dialog.

Fixes #25676.

crisbeto committed:
fix(material/bottom-sheet): aria-modal not being set

Fixes that `aria-modal` wasn't being set on the bottom sheet container. This likely broke when the bottom sheet was refactored to use the CDK dialog.

andrewseguin committed:
fix(material/core): add typography to prebuilt themes (#25696)

crisbeto committed:
fix(material/tooltip): animations running when timeouts haven't elapsed (#25699)

After the tooltips were switched to CSS animations, a regression was introduced where the opposite animation is shown even if the tooltip didn't actually reach its target state.

These changes are an alternate take on the fix from #24652 which had to be reverted due to internal failures.

Fixes #24614.

devversion committed:
docs: fix double `will` in list mdc migration snippet (#25704)

Also kept the Prettier formatting recommendations-omitting the trailing
whitespace. Should be good to keep.

andrewseguin committed:
feat(material/form-field): add error harness (#25698)

* feat(material/form-field): add error harness

* feat(material/form-field): deprecate legacy error harness

* fix(material/form-field): use error harness for all getters

spiralx committed:
docs(material/input): Fix cdkTextareaAutosize link (#25688)

The link to the `cdkTextareaAutosize` directive in the "Auto-resizing <textarea> elements" section on the overview page links to that section on the same page, have changed it to correctly point to that section within the `cdk/text-field` overview page.

crisbeto committed:
docs(material/select): update MDC migration guide

Updates the MDC migration guide with the select changes. Also adds some of the same changes to the autocomplete since a lot of the code is shared.

crisbeto committed:
docs(material/radio): update MDC migration guide

Updates the MDC migration guide to reflect the changes to the radio button.

crisbeto committed:
docs(material/menu): update MDC migration guide

Updates the MDC migration guide with the menu changes.

crisbeto committed:
docs(material/tabs): update tab nav panel changes in MDC migration doc

Resolves a TODO in the MDC migration doc about elaborating further on the new a11y pattern for the tab nav bar.

andrewseguin committed:
feat(material/core): add option to disable core typography (#25706)

mmalerba committed:
feat(material/datepicker): make compatible with MDC & legacy components (#25648)

BREAKING CHANGE:
Buttons inside the datepicker popup and datepicker toggle now use the
MDC-based button implementation. They have different CSS classes and
styles, so custom style overrides may need to be updated

andrewseguin committed:
docs(material/table): update MDC changes (#25717)

andrewseguin committed:
docs(material/tooltip): update MDC changes (#25718)

* docs(material/tooltip): update MDC changes

* docs(material/tooltip): remove font-size

* docs(material/tooltip): fix font-size

* docs(material/tooltip): fix wording

zarend committed:
docs(material/chips): migrate examples to MDC (#25661)

Migrate chips examples to use mdc-based components. Add documentation
overview for mdc-based chips component.

zarend committed:
docs: release notes for the v14.2.3 release

zarend committed:
release: cut the v15.0.0-next.2 release

crisbeto committed:
fix(cdk/a11y): clean up list key manager on destroy (#25715)

Historically we haven't needed to do any cleanup for the `ListKeyManager`, because it didn't contain any rxjs subscriptions. Over time we've accumulated more functionality which now needs to be cleaned up.

These changes introduce a `destroy` method and calls it in all the relevant places. I also removed any places where we previously had to do manual cleanup.

Fixes #25537.

mmalerba committed:
fix(material/schematics): add explicit typography @include in v15 update (#25724)

andrewseguin committed:
feat(material/core): remove typography styles from core mixin (#25723)

* feat(material/core): remove typography styles from core mixin

* fix(material/core): remove import

* fix(material/core): update comment

mmalerba committed:
release: cut the v15.0.0-next.3 release

Commit:
fix(material/button): cdk-focus classes not being applied (#25619)

Fixes a bug in the Angular Material `button` component where
cdk-focused, cdk-keyboard-focused, cdk-mouse-focused were
not being applied upon respective focus. This is because
there was no FocusManager set up with the MDC buttons like
the legacy button did.

Fixes #25618

mmalerba committed:
fix(material/schematics): fix calling .match on null

zarend committed:
fix(material/list): visually indicate active links in HCM (#25679)

For anchor tags inside of a list, draw a circle in HCM mode to indicate
the active link item. This fixes an a11y issue where active list-items
had no visual indication of being active in HCM.

mmalerba committed:
test(material/schematics): update integration test

The test should reflect running `ng generate` on an app that has already
done `ng update` to v15 and is using legacy components.

mmalerba committed:
test(material/schematics): add package.json to golden project

mmalerba committed:
fix(material/schematics): correctly migrate theme, color & typography

crisbeto committed:
refactor(material/slider): switch to new theming API (#25725)

Switches the MDC-based slider to use the new theming API based on CSS variables.

andrewseguin committed:
fix(material/core): remove core style parameters (#25738)

* fix(material/core): remove core style parameters

* fix(material/core): dev-app theme

andrewseguin committed:
fix(material/core): add typography hierarchy to prebuilt (#25746)

crisbeto committed:
fix(material/schematics): files with inline resources overwritten by ng-generate (#25744)

The `ng generate` MDC migration schematics was assuming that `visitTemplate` and `visitStylesheet` will be called only on external resources and it was overwriting the entire file content which was breaking files that have components with inline resources.

These changes edit the file only within the template/stylesheet span.

crisbeto committed:
fix(material/tooltip): animation not cancelled when mouseleave goes through tooltip (#25740)

We ignore `mouseleave` events that go into the tooltip which meant that some animations weren't being cancelled correctly after the fix in #25699. These changes resolve the issue by skipping all animations if the tooltip isn't fully shown yet.

Fixes #24614.

crisbeto committed:
fix(material/tooltip): resolve server-side rendering error (#25728)

In a recent change MDC introduced a top-level reference to `window` which broke our server-side rendering check. These changes remove our dependency on MDC completely in order to fix the error.

crisbeto committed:
fix(material/chips): aria-selected not reflecting selection state (#25742)

When the chip listbox is in single selection mode, it needs to remove `aria-selected` from the selected option and move it to the newly-selected one. The listbox was doing this already, but change detection wasn't being triggered which meant that the DOM was out of date.

zarend committed:
fix(material/list): communicate current active page with aria-current (#25681)

For listitems that are anchor tags, communicate the activated state to
screen readers using aria-current. Add aria-current="page" to listitems
that are both activated and anchor tags. Fix a11y issue where the
activated state is not communicated to screen readers.

mmalerba committed:
fixup! test(material/schematics): add package.json to golden project

titosx committed:
docs(material/theming): fix typo in theming guide (#25685)

In theming guide the mat.get-color-from-palette is shown with a number identifier in quotes, unlike the theming-your-components guide which shows it without quotes.

Fixes angular#25598

josephperrott committed:
build: update saucelabs connect version (#25636)

Update to saucelabs connect 4.8.1

andrewseguin committed:
docs(material/core): update typography guide (#25745)

crisbeto committed:
fix(material/schematics): replace pre-existing attribute values (#25754)

Fixes that the MDC migration schematic was always adding new attributes, instead of replacing the values of the ones that are already there.

andrewseguin committed:
docs(material/tabs): update MDC changes (#25760)

mmalerba committed:
fix(material/schematics): ensure test files are included in migration (#25757)

wagnermaciel committed:
docs: release notes for the v14.2.4 release

wagnermaciel committed:
release: cut the v15.0.0-next.4 release

mmalerba committed:
fix(material/schematics): gracefully skip broken css files (#25767)

jokecamp committed:
fix(cdk/drag-drop): native event not passed as parameter to drop container (#25763)

* fix(cdk/drag-drop): native event not passed as parameter to drop CredentialsContainer

Fixes a bug where native event was always an empty object because it was not passed to the container drop function and the default `{}` object was always used.

* fix(cdk/drag-drop): Added missing trailing comma

crisbeto committed:
feat(material/button): add variant to button harness (#25770)

Adds the ability to get the variant of a button harness and to filter by it.

Fixes #25755.

zarend committed:
fix(material/chips): don't remove aria-selected from deselected options (#25748)

For chip-listbox, set `aria-selected="false"` on deselected options that
are selectable. Conforms with [WAI ARIA Listbox authoring practices guide](
https://www.w3.org/WAI/ARIA/apg/patterns/listbox/), which says to always
include aria-selected attribute on options that are selectable. Fix issue where
voiceover reads every option as "selected" (#25736).

Fix #25736

crisbeto committed:
docs(material/badge): don't use inline elements in badge demo (#25758)

The badge docs were saying that badges might not work on inline elements while the example was still using inline elements itself. These changes update the example to use `div` elements instead.

Fixes #25751.

crisbeto committed:
refactor: remove workaround for TS versions earlier than 4.8 (#25777)

Cleans up some workarounds that were put in place to support version of TypeScript earlier than 4.8.

crisbeto committed:
fix(material/schematics): rename references in MDC generate schematic (#25773)

Currently the `ng generate` migration that migrates from the legacy symbols to the non-legacy ones renames references, but only in some specific cases (e.g. `imports` of an `NgModule`). These changes expand the migration to cover any reference within the file. I've also tried to reduce the amount of migration data that we need to maintain.

angular-robot committed:
build: update angular shared dev-infra code to a96fc0c (#25639)

crisbeto committed:
fix(material/icon): forward fontIcon attribute to element (#25779)

In #24578 we added the ability to set font icons using the `fontIcon` input, however it was never forwarded to the `fontIcon` attribute which means using a data binding won't work.

Fixes #25759.

angular-robot committed:
build: update angular shared dev-infra code to 9dcbebb

josephperrott reviewed (approved):
LGTM
